src/heimdall/worker.cc

```diff
@@ -0,0 +1,903 @@
+                              vtzero::encoded_property_value(node->tagged_access()));
+    node_feature.add_property(key_private_access_,
+                              vtzero::encoded_property_value(node->private_access()));
+    node_feature.add_property(key_cash_only_toll_,
+                              vtzero::encoded_property_value(node->cash_only_toll()));
+    node_feature.add_property(key_mode_change_, vtzero::encoded_property_value(node->mode_change()));
+    node_feature.add_property(key_named_intersection_,
+                              vtzero::encoded_property_value(node->named_intersection()));
+    node_feature.add_property(key_is_transit_, vtzero::encoded_property_value(node->is_transit()));
+    node_feature.add_property(key_transition_count_,
+                              vtzero::encoded_property_value(node->transition_count()));
+
+    node_feature.commit();
+  }
+
+private:
+  vtzero::layer_builder layer_;
+
+  // Pre-registered keys
+  vtzero::index_value key_tile_level_;
+  vtzero::index_value key_tile_id_;
+  vtzero::index_value key_node_id_;
+  vtzero::index_value key_node_type_;
+  vtzero::index_value key_traffic_signal_;
+  // Individual access mode keys
+  vtzero::index_value key_access_auto_;
+  vtzero::index_value key_access_pedestrian_;
+  vtzero::index_value key_access_bicycle_;
+  vtzero::index_value key_access_truck_;
+  vtzero::index_value key_access_emergency_;
+  vtzero::index_value key_access_taxi_;
+  vtzero::index_value key_access_bus_;
+  vtzero::index_value key_access_hov_;
+  vtzero::index_value key_access_wheelchair_;
+  vtzero::index_value key_access_moped_;
+  vtzero::index_value key_access_motorcycle_;
+  vtzero::index_value key_edge_count_;
+  vtzero::index_value key_intersection_;
+  vtzero::index_value key_density_;
+  vtzero::index_value key_local_edge_count_;
+  vtzero::index_value key_drive_on_right_;
+  vtzero::index_value key_elevation_;
+  vtzero::index_value key_tagged_access_;
+  vtzero::index_value key_private_access_;
+  vtzero::index_value key_cash_only_toll_;
+  vtzero::index_value key_mode_change_;
+  vtzero::index_value key_named_intersection_;
+  vtzero::index_value key_is_transit_;
+  vtzero::index_value key_transition_count_;
+};
+
+} // anonymous namespace
+
+void heimdall_worker_t::ReadZoomConfig(const boost::property_tree::ptree& config) {
+  min_zoom_road_class_ = kDefaultMinZoomRoadClass;
+
+  auto tile_config = config.get_child_optional("tile");
+  if (tile_config) {
+    auto zoom_array = tile_config->get_child_optional("min_zoom_road_class");
+    if (zoom_array) {
+      size_t i = 0;
+      for (const auto& item : *zoom_array) {
+        if (i < kNumRoadClasses) {
+          min_zoom_road_class_[i] = item.second.get_value<uint32_t>();
+          ++i;
+        }
+      }
+      // Fill remaining values with defaults if array is too short
+      for (; i < kNumRoadClasses; ++i) {
+        min_zoom_road_class_[i] = kDefaultMinZoomRoadClass[i];
+      }
```

Comment:
IMO we should error here. we can't really assume people omitted only the last ones, could be any. if they set this config, they should give us all values